src/test/kotlin/com/atlassian/performance/tools/virtualusers/SimpleWebdriverScenario.kt

```diff
@@ -0,0 +1,84 @@
+package com.atlassian.performance.tools.virtualusers
+
+import com.atlassian.performance.tools.jiraactions.api.ActionType
+import com.atlassian.performance.tools.jiraactions.api.WebJira
+import com.atlassian.performance.tools.jiraactions.api.action.Action
+import com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter
+import com.atlassian.performance.tools.jiraactions.api.memories.User
+import com.atlassian.performance.tools.virtualusers.api.browsers.CloseableRemoteWebDriver
+import com.atlassian.performance.tools.virtualusers.api.browsers.HeadlessChromeBrowser
+import com.atlassian.performance.tools.virtualusers.api.config.VirtualUserTarget
+import com.atlassian.performance.tools.virtualusers.api.diagnostics.WebDriverDiagnostics
+import com.atlassian.performance.tools.virtualusers.lib.api.Scenario
+import java.util.concurrent.atomic.AtomicInteger
+
+// TODO show that RTE can be disabled via setup
+class SimpleWebdriverScenario(
+    virtualUserTarget: VirtualUserTarget,
+    private val meter: ActionMeter
+) : Scenario(
+    virtualUserTarget, meter
+) {
+    private val driver: CloseableRemoteWebDriver = HeadlessChromeBrowser().start()
+    private val jira: WebJira = WebJira(
+        driver = driver.getDriver(),
+        base = virtualUserTarget.webApplication,
+        adminPassword = "admin"
+    )
+
+    override fun before() { //todo don't use hardcoded credentials
+        jira.goToLogin().logIn(User(
+            "admin",
+            "admin"
+        ))
+    }
+
+    override fun cleanUp() {
+        driver.close()
+    }
+
+    override fun getActions(): List<Action> {
```

Comment:
We need a pattern for partially reusable/customizable scenarios. E.g. reuse this `getActions`, but with a different `Browser`.
I imagine, we could inject a `Browser` like this:
```
open class SimpleWebdriverScenario(
    private val browser: Browser,
    private val target: VirtualUserTarget,
    private val meter: ActionMeter
) : Scenario(
    target,
    meter
) {
    
    constructor(
        target: VirtualUserTarget,
        meter: ActionMeter
    ) : this(
        HeadlessChromeBrowser(),
        target,
        meter
    )
}

class CustomizedWebdriverScenario(
    private val target: VirtualUserTarget,
    private val meter: ActionMeter
) : SimpleWebdriverScenario(
    FirefoxBrowser(),
    target,
    meter
)
```

Alternatively we could extract each method into a "util" and let the user stitch a new scenario class from these components.
WDYT @wyrzyk?

Reply:
We can support the injection of the browser, but it makes API focused on one load generator type. The new VUB builder will have to support Browser parameter which will be only usable in this kind of scenarios. 

If we look at how the browser has been changed during the last years in our scenarios/tests, it was mainly bumping the version and one switch from chrome to chromium. We never needed the flexibility of decoupled load generator and scenario. 

Beside generic scenarios, we provide in jira-actions and jira-software-actions scenario is part of the test code. We have a couple of different scenarios in our internal tests. For our internal tests, it's even easier to create an instance of a browser instead of passing it via the builder.

It also makes testing scenarios easier because the class provides all the capabilities to perform the test. All that needs to be provided is VUT and action meter.

If we really want to provide a way to reuse scenario with a different browser, then it doesn't have to be simple. It's not the main or common use case. We can put all the responsibility to provide the cross-browser scenario on the client's code. We could just provide passing properties to VU. It could look like:

```
 class SimpleCrossBrowserScenario(
        private val target: VirtualUserTarget,
        private val meter: ActionMeter
    ) : Scenario(
        target,
        meter
    ) {
        val driver = SystemPropertyDriverFactory().getDriver()
    }

    class SystemPropertyDriverFactory {
        fun getDriver(): WebDriver {
            return when (System.getProperty("browser")) {
                "firefox" -> new Firefox ()
                "chrome" -> new Chrome ()
            }
        }
    }
```

1. It removes the mental overhead of browser from the basic usages of VUB.
2. Creator of the scenario controls supported browsers/load generators (just by using a common webdriver API there's no guarantee the scenario will run with all the browsers and all the configurations).  
3. It's up to the creator of the scenario to provide API (properties needed to configure load generator).
4. It can be used for different's kind of generators. For example to provide for HTTP based generators on HttpClient and OkHttp. For DB based generators (different JDBC implementations) etc.

Reply:
I didn't mention VUB or VUT. Did you see my code snippet suggestion?

Reply:
Ok, I see now. We always use 2 parameters constructor, so there's no VUB or VUT problem. This way it's easier to use different browsers, but still, we don't have evidence that the feature is needed.

It's also more prone to errors. It's hard to enforce inheritance only the specific way.
This will not work as expected:
```
class CustomizedWebdriverScenario(
    private val browser: Browser,
    private val target: VirtualUserTarget,
    private val meter: ActionMeter
) : SimpleWebdriverScenario(
   browser,
    target,
    meter
)
```

Reply:
Yep, it could be a WTF moment.
> It's also more prone to errors.

More prone to errors in comparison to `SystemPropertyDriverFactory`?
How would we pass the properties to the VU JVMs?

Reply:
It's more prone to errors because it adds WTF moment to the basic scenario implementation journey instead of adding WTF moment to an imaginary scenario without a real need. 

> More prone to errors in comparison to SystemPropertyDriverFactory?
> How would we pass the properties to the VU JVMs?

`SystemPropertyDriverFactory` is just one example of implementation. We can pass params via VUB (command line parameters).

We can also use the `properties object` as a 3rd parameter to the scenario. It would be passed via VUB to the scenario and will allow scenarios to implement own API (for example. load generators support, credentials support).